.github/workflows/generator_container_slsa3.yml

```diff
@@ -74,6 +91,8 @@ jobs:
   # with the release at the ref detected by `detect-env`. This must be a tag
   # reference.
   generator:
```

Comment:
the last job in the workflow (`final`) always runs using `if: always()` and collects all the `outcome`s from the jobs it depends on. So if `generator` job is skipped because of a an error of a previous job, the final job will still collect at least one of the failing job's `outcome` and fail accordingly. The `generator` job's `outcome` would be empty, but the previous job's won't be so `final` will still find the failure. 

Let me know if this makes sense or if you think there's a logic error.

Reply:
My question was that wouldn't `generator` still attempt to run if `detect-env` had a `continue-on-error`? Maybe the `needs` was actually the key (is `needs` requiring that the previous job succeeds?)

Reply:
Got it: yes, `needs` requires success https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds